addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2686,7 +2686,28 @@ const SnippetOptionWidget = Widget.extend({
 
         // Ask a mutexed snippet update according to the widget value change
         const shouldRecordUndo = (!previewMode && !ev.data.isSimulatedEvent);
+        if (!widget.actionQueue) {
+            widget.actionQueue = [];
+        }
+        const currentAction = {previewMode};
+        widget.actionQueue.push(currentAction);
         this.trigger_up('snippet_edition_request', {exec: async () => {
+            widget.actionQueue = widget.actionQueue.filter((action, i, actions) => {
+                const prev = actions[i - 1];
+                const next = actions[i + 1];
+                if (action.previewMode === true && next && next.previewMode === 'reset') {
+                    return false;
+                } else if (action.previewMode === 'reset' && prev && prev.previewMode === true) {
+                    return false;
+                }
```

Comment:
I went with the safest removals only and it already improves the UX a lot. While your suggestions make sense I think these happen either very rarely or never in practice: a widget that gets previewed always gets reset before it gets previewed again, so the result is the same whether we ignore all previews/resets before the last or not. And for the second case, it basically only happens if the user activates the same option twice in a row. I don't know if it would make the code easier to read and reason about, or if it might introduce some non-obvious side-effects but I'm not opposed to those changes in principle :)

Reply:
> a widget that gets previewed always gets reset before it gets previewed again

Not with a we-input, there are no resets at all ;) Leave a key pressed and you will have 1000 queued previewMode=true ;)

I guess we can go with my first suggestion since it reduces the code but stay safer by not implementing the second suggestion. Difficult to be entirely sure of everything here :/ I guess time will tell.

Reply:
Well I tried both of your suggestions but the issue with this approach is that it cannot completely collapse an action list, it will always keep the last item. I can relax the restriction on the type of preview on the left of resets and on the right of previews, and this would be functionally equivalent to only keeping the last preview/reset in a series of actions of the same type, and then collapsing preview > reset pairs. This will work correctly for inputs by skipping intermediate previews, and will still completely skip actions if the entire list can be collapsed to nothing.

I'm also worried that some options could not perform a reset correctly without having had a preview first, since they may store the current state on preview so that they can reset it later, but if they never had a preview there would be no such state.